libs/Database.js

```diff
@@ -531,7 +639,6 @@ class Database {
             await this.flushCache();
           }
         }
-        const finalTableName = `${contract}_${table}`;
 
         const tableData = await this.getContractCollection(contract, finalTableName);
```

Comment:
same thing here, can avoid doing findContract twice if getContractCollection also returns the primaryKey